ca_gazebo/src/color_sensor_plugin.cpp

```diff
@@ -0,0 +1,131 @@
+#include <gazebo/common/Plugin.hh>
+#include <ros/ros.h>
+#include "ca_gazebo/color_sensor_plugin.h"
+
+#include "gazebo_plugins/gazebo_ros_camera.h"
+
+#include <string>
+
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/CameraSensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+
+
+#include "std_msgs/Bool.h"
+
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_SENSOR_PLUGIN(GazeboRosColor)
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Constructor
+  GazeboRosColor::GazeboRosColor():
+  _nh("color_sensor_plugin"),
+  _fov(6),
+  _range(10)
+{
+
+}
+  ////////////////////////////////////////////////////////////////////////////////
+  // Destructor
+  GazeboRosColor::~GazeboRosColor()
+  {
+    ROS_DEBUG_STREAM_NAMED("camera","Unloaded");
+  }
+
+  void GazeboRosColor::Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+
+    CameraPlugin::Load(_parent, _sdf);
+    // copying from CameraPlugin into GazeboRosCameraUtils
+    this->parentSensor_ = this->parentSensor;
+    this->width_ = this->width;
+    this->height_ = this->height;
+    this->depth_ = this->depth;
+    this->format_ = this->format;
+    this->camera_ = this->camera;
+    this->_rvalue_max = _sdf->GetElement("rvalue_max")->Get<double>();
```

Comment:
This is not the way of getting the parameters. Use the templetized methods. For example:

```C++
this->_rvalue_max = _sdf->GetElement<double>("rvalue_max");
```

Reply:
This gives me the following error:
```
/create_ws/src/create_autonomy/ca_gazebo/src/color_sensor_plugin.cpp:56:42: err$
r: expected primary-expression before ‘double’
     this->rvalue_max_ = _sdf->GetElement<double>("rvalue_max");
```
The info that I found says that Element was a class of Gazebo in the earlier versions, but I couldn't find how to use it in the 9.0 API, the old one however does follow the format that I was using: [http://osrf-distributions.s3.amazonaws.com/gazebo/api/1.3.0/classsdf_1_1Element.html#a29fa74bb6bf29701e82b6353e732d31d](url)

Reply:
Erasing the "Element" part apparently solves the issue

`this->rvalue_max_ = _sdf->Get<double>("rvalue_max");`